### ISSUE TITLE
Custom views broken after v5.9.4 release

### ISSUE THREAD

**Issue by stepanataccolade** [Shippable/support#3861]
### Description of your issue:
After today's release, my custom views are showing jobs that should not be part of it.
My view is defined to use:
- one ci project `konciergeMD/shippable-pipeline-scripts`
- one flag - `shippable-pipeline-scripts-tests`

But I'm seeing jobs there, that are flagged with different flags and are coming from different repositories of our organization.  

Link to the view:
https://app.shippable.com/accounts/5798717a413c350c00d7af06/dashboard/7848

![image](https://user-images.githubusercontent.com/19684788/30975089-2ecce84a-a426-11e7-84ee-48a2102d36d0.png)
Previously only the pipeline in the middle that continues to the right from the view has been visible

**Comment by manishas** [comment]
@ragesh @avinci can you please follow up on this?

**Comment by avinci** [comment]
@stepanataccolade TLDR - I think there is some issue with filtering as we moved it upstream. I will get back to you on what the problem is. 

Long version
I think there is something missing from our filtering process. We are working on performance fixes to make SPOG load fast, and one of the key problems was how flags were being evaluated. We were filtering way to late in the cycle and so your SPOG was being very slow due to this. I.e. filter and paint as opposed to we were painting and filtering.

Since the filtering logic was moved upstream, it has caused this behavior. Need to look into what the exact cause is. I looked at it real quick, it has got to do with the repo that is being shared with the unnecessary objects and the flagged objects...

I will have an update soon.
